browser/src/layer/tile/CalcTileLayer.js

```diff
@@ -402,25 +402,25 @@ L.CalcTileLayer = L.CanvasTileLayer.extend({
 
 		const mapElement = document.getElementById('map'); // map's size = tiles section's size.
 		const oldMapSize = [mapElement.clientWidth, mapElement.clientHeight];
+		const widthIncreased = oldMapSize[0] < newMapSize[0];
+		const heightIncreased = oldMapSize[1] < newMapSize[1];
+		const sizeChanged = oldMapSize[0] !== newMapSize[0] || oldMapSize[1] !== newMapSize[1];
 
 		// Early exit. If there is no need to update the size, return here.
-		if (oldMapSize[0] === newMapSize[0] && oldMapSize[1] === newMapSize[1])
-			return false;
-		else {
+		if (sizeChanged) {
 			this._resizeMapElementAndTilesLayer(mapElement, marginLeft, marginTop, newMapSize);
 
-			const widthIncreased = oldMapSize[0] < newMapSize[0];
-			const heightIncreased = oldMapSize[1] < newMapSize[1];
-
-			this._map.invalidateSize({}, new L.Point(oldMapSize[0], oldMapSize[1]));
+			this._map.invalidateSize(false, new L.Point(oldMapSize[0], oldMapSize[1]));
 			app.sectionContainer.onResize(newCanvasSize[0], newCanvasSize[1]); // Canvas's size = documentContainer's size.
 			this._updateHeaderSections();
 
 			this._mobileChecksAfterResizeEvent(heightIncreased);
+		}
 
-			// Center the view w.r.t the new map-pane position using the current zoom.
-			this._map.setView(this._map.getCenter());
+		// Center the view w.r.t the new map-pane position using the current zoom.
+		this._map.setView(this._map.getCenter());
```

Comment:
if size is not changed - do we want to call `setView` ?
I know it does a scroll/pan/zoom as comment says

Reply:
Yes we need to set the view after a position change. There is a new size and a new center now.